tests/test_07_l2vpn_return_codes.py

```diff
@@ -0,0 +1,256 @@
+        payload = {
+            "name": "Test L2VPN request",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "500"},
+                {"port_id": "urn:sdx:port:sax.net:Sax01:50","vlan": "500"}
+            ]
+        }
+        response = requests.post(api_url, json=payload)
+        assert response.status_code == 201, response.text
+
+        # edit the first l2pvn to match the newly created one
+        self.payload['endpoints'][0]['vlan'] = "500"
+        self.payload['endpoints'][0]['vlan'] = "500"
+        response = requests.patch(f"{api_url}/{self.key}", json=self.payload)
+        assert response.status_code == 409, response.text
+
+    def test_060_edit_l2vpn_with_min_bw_out_of_range_code410(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        410: Can't fulfill the strict QoS requirements
+        Case: min_bw out of range (value must be in [0-100])
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        self.payload['qos_metrics'] = {'min_bw':{'value': 101}}
+        response = requests.patch(f"{api_url}/{self.key}", json=self.payload)
+        assert response.status_code == 410, response.text
+
+    @pytest.mark.xfail(reason="return status 400: PCE error: Can't find a valid vlan breakdown solution")
+    def test_061_edit_l2vpn_with_max_delay_out_of_range_code410(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        410: Can't fulfill the strict QoS requirements
+        Case: max_delay out of range (value must be in [0-1000])
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        self.payload['qos_metrics'] = {'max_delay':{'value': 1001}}
+        response = requests.patch(f"{api_url}/{self.key}", json=self.payload)
+        assert response.status_code == 410, response.text
+
+    @pytest.mark.xfail(reason="return status 400: PCE error: Can't find a valid vlan breakdown solution")
+    def test_062_edit_l2vpn_with_max_number_oxps_out_of_range_code410(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        410: Can't fulfill the strict QoS requirements
+        Case: max_number_oxps out of range (value must be in [0-100])
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        self.payload['qos_metrics'] = {'max_number_oxps':{'value': 101}}
+        response = requests.patch(f"{api_url}/{self.key}", json=self.payload)
+        assert response.status_code == 410, response.text
+           
+    @pytest.mark.xfail(reason="return status 400: PCE error: Can't find a valid vlan breakdown solution")
+    def test_072_edit_l2vpn_with_impossible_scheduling_code411(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        411: Scheduling not possible
+        end_time before current date
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        self.payload['scheduling'] = {'end_time': "2023-12-30"}
```

Comment:
date/time format is wrong.

Reply:
The format validation is failing. It works fine with the format "YYYY-MM-DD", but not with "YYYY-MM-DDTHH:mm:SSZ".